iroh-bytes/src/provider.rs

```diff
@@ -155,6 +155,17 @@ pub enum AddProgress {
 /// Progress updates for the get operation.
 #[derive(Debug, Serialize, Deserialize)]
 pub enum DownloadProgress {
+    /// Data was found locally.
+    FoundLocal {
+        /// child offset
+        child: u64,
+        /// The hash of the entry.
+        hash: Hash,
+        /// The size of the entry in bytes.
+        size: u64,
```

Comment:
is this the full size, or the size that resides locally?

Reply:
No, it is the full size. That is one thing we know as soon as we get any data. To compute the size that is local you need to use this and valid_ranges.

Reply:
I can see how the ranges could be useful, if you want to show a visualization which pieces of a file are already available on disk. But the more common use case seems to be just showing `x/y` has already been downloaded.

Reply:
so maybe at least have a simple helper method on getting the local size?